msvc/tools-build.targets

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="GenVersionFiles" AssemblyFile="GenVersionFiles.dll"/>
+
+  <PropertyGroup>
+    <WinObjCVersionFile>$([System.IO.Path]::Combine('$(MSBuildThisFileDirectory)', '..\build\winobjc.version'))</WinObjCVersionFile>
+    <SharedVersionHeaderDir>$(IntermediateOutputPath)</SharedVersionHeaderDir>
+    <SharedHeaderFile Condition="'$(ProjectExt)' == '.csproj'">$(SharedVersionHeaderDir)\SharedAssemblyInfo.cs</SharedHeaderFile>
+    <SharedHeaderFile Condition="'$(ProjectExt)' == '.vcxproj'">$(SharedVersionHeaderDir)\SharedVersion.h</SharedHeaderFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <BuildDependsOn Condition="'$(ProjectExt)' == '.csproj'">GenSharedAssemblyFile;$(BuildDependsOn);</BuildDependsOn>
+    <BuildDependsOn Condition="'$(ProjectExt)' == '.vcxproj'">GenSharedHeaderFile;$(BuildDependsOn);</BuildDependsOn>
+  </PropertyGroup>
+
+  <Target Name="MakeSharedVersionHeaderDir" >
+    <MakeDir Directories="$(SharedVersionHeaderDir)" />
+  </Target>
+
+  <Target
+    Name="GenSharedAssemblyFile"
+    Inputs="$(WinObjCVersionFile)"
```

Comment:
>WinObjCVersionFile [](start = 14, length = 18)

I'm a slightly bit concerned that each tools binary is going to rebuild the version file.  This means that each binary will always be out-of-date.
However,  given our expected use cases , I think this is acceptable.